README.md

```diff
@@ -1,8 +1,361 @@
 # CUDA C++ Core Libraries (CCCL)
 
-The future home of the CUDA C++ Core Libraries (Thrust, CUB, libcu++).
+Welcome to the CUDA C++ Core Libraries (CCCL) where our mission is to make CUDA C++ more delightful.
+
+This repository unifies three essential CUDA C++ libraries into a single, convenient repository: 
+
+- [Thrust](thrust) (former [repo](https://github.com/nvidia/thrust))
+- [CUB](cub) (former [repo](https://github.com/nvidia/cub))
+- [libcudacxx](libcudacxx) (former [repo](https://github.com/nvidia/libcudacxx))
+
+The goal of CCCL is to provide CUDA C++ developers with building blocks that make it easier to write safe and efficient code. 
+Bringing these libraries together streamlines your development process and broadens your ability to leverage the power of CUDA C++.
+For more information about the decision to unify these projects, see the announcement here: (TODO)
+
+## Overview
+
+The concept for the CUDA C++ Core Libraries (CCCL) grew organically out of the Thrust, CUB, and libcudacxx projects that were developed independently over the years with a similar goal: to provide high-quality, high-performance, and easy-to-use C++ abstractions for CUDA developers.
+Naturally, there was a lot of overlap among the three projects, and it became clear the community would be better served by unifying them into a single repository.
+
+- **Thrust** is the C++ parallel algorithms library which inspired the introduction of parallel algorithms to the C++ Standard Library. Thrust's high-level interface greatly enhances programmer productivity while enabling performance portability between GPUs and multicore CPUs via configurable backends that allow using multiple parallel programming frameworks (such as CUDA, TBB, and OpenMP).
+
+- **CUB** is a lower-level, CUDA-specific library designed for speed-of-light parallel algorithms across all GPU architectures. In addition to device-wide algorithms, it provides *cooperative algorithms* like block-wide reduction and warp-wide scan, providing CUDA kernel developers with building blocks to create speed-of-light, custom kernels. 
+
+- **libcudacxx** is the CUDA C++ Standard Library. It provides an implementation of the C++ Standard Library that works in both host and device code. Additionally, it provides abstractions for CUDA-specific hardware features like synchronization primitives, cache control, atomics, and more. 
+
+The main goal of CCCL is to fill a similar role that the Standard C++ Library fills for Standard C++ – provide general-purpose, speed-of-light tools to CUDA C++ developers with general-purpose, speed-of-light tools to focus on solving the problems that matter.
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**💡 suggestion:** ‏ In addition to @senior-zero 's suggestion here, use a colon instead of a hyphen (if you want a dash, use an em-dash, not a hyphen). And fill in some missing verbs.

```suggestion
The main goal of CCCL is to fill a similar role that the Standard C++ Library fills for Standard C++: to provide general-purpose, speed-of-light tools to enable CUDA C++ developers to focus on solving the problems that matter.
```